atr/shared/distribution.py

```diff
@@ -36,55 +35,64 @@
-    version = forms.string("Version", placeholder="E.g. 1.2.3, without a leading v")
-    details = forms.checkbox("Include details", description="Include the details of the distribution in the response")
-    submit = forms.submit("Record distribution")
-
-    async def validate(self, extra_validators: dict | None = None) -> bool:
-        if not await super().validate(extra_validators):
-            return False
-        if not self.platform.data:
-            return False
-        default_owner_namespace = self.platform.data.value.default_owner_namespace
-        requires_owner_namespace = self.platform.data.value.requires_owner_namespace
-        owner_namespace = self.owner_namespace.data
-        # TODO: We should disable the owner_namespace field if it's not required
-        # But that would be a lot of complexity
-        # And this validation, which we need to keep, is complex enough
-        if default_owner_namespace and (not owner_namespace):
-            self.owner_namespace.data = default_owner_namespace
-        if requires_owner_namespace and (not owner_namespace):
-            msg = f'Platform "{self.platform.data.name}" requires an owner or namespace.'
-            return forms.error(self.owner_namespace, msg)
-        if (not requires_owner_namespace) and (not default_owner_namespace) and owner_namespace:
-            msg = f'Platform "{self.platform.data.name}" does not require an owner or namespace.'
-            return forms.error(self.owner_namespace, msg)
-        return True
-
-
-@dataclasses.dataclass
-class FormProjectVersion:
-    form: DistributeForm
-    project: str
-    version: str
+
+    @pydantic.field_validator("platform", mode="before")
+    @classmethod
+    def validate_platform(cls, value: Any) -> sql.DistributionPlatform:
+        if isinstance(value, sql.DistributionPlatform):
+            return value
+        if isinstance(value, str):
+            try:
+                return sql.DistributionPlatform[value]
+            except KeyError:
+                raise ValueError(f"Invalid platform: {value}")
+        raise ValueError(f"Platform must be a string or DistributionPlatform, got {type(value)}")
+
+    @pydantic.model_validator(mode="after")
+    def validate_owner_namespace(self):
+        if not self.platform:
+            raise ValueError("Platform is required")
+
+        default_owner_namespace = self.platform.value.default_owner_namespace
+        requires_owner_namespace = self.platform.value.requires_owner_namespace
+
+        # Set default if needed and not provided
+        if default_owner_namespace and not self.owner_namespace:
+            self.owner_namespace = default_owner_namespace
+
+        # Validate requirements
+        if requires_owner_namespace and not self.owner_namespace:
+            raise ValueError(f'Platform "{self.platform.name}" requires an owner or namespace.')
+
+        if not requires_owner_namespace and not default_owner_namespace and self.owner_namespace:
```

Comment:
Please ensure that all subexpressions are parenthesized, as pointed out in our [code conventions](https://release-test.apache.org/docs/code-conventions). Same goes for prior lines in this method, and potentially elsewhere.